src/vs/editor/contrib/inlineEdit/browser/inlineEditController.ts

```diff
@@ -121,11 +127,14 @@ export class InlineEditController extends Disposable {
 		}));
 
 		//Invoke provider on focus
-		this._register(editor.onDidFocusEditorText(async () => {
-			if (!this._enabled.get()) {
+		const editorFocusSignal = observableSignalFromEvent('InlineEditController.editorFocusSignal', editor.onDidFocusEditorText);
```

Comment:
For focus, I can recommend to have an `isFocused: IObservable<boolean>`, instead of having two signals indicating the state transitions.